src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts

```diff
@@ -319,14 +318,15 @@ class CodeActionOnSaveParticipant implements ITextFileSaveParticipant {
 		if (!codeActionsOnSave.length) {
 			return undefined;
 		}
-
-		const excludedActions = Object.keys(setting)
-			.filter(x => convertedSetting[x] === 'never' || false)
-			.map(x => new CodeActionKind(x));
+		const excludedActions = Array.isArray(setting)
+			? []
+			: Object.keys(setting)
+				.filter(x => setting[x] === 'never' || false)
+				.map(x => new CodeActionKind(x));
 
 		progress.report({ message: localize('codeaction', "Quick Fixes") });
 
-		const filteredSaveList = codeActionsOnSave.filter(x => convertedSetting[x.value] === 'always' || (convertedSetting[x.value] === 'explicit') && env.reason === SaveReason.EXPLICIT);
+		const filteredSaveList = Array.isArray(setting) ? codeActionsOnSave : codeActionsOnSave.filter(x => setting[x.value] === 'always' || ((setting[x.value] === 'explicit' || true) && env.reason === SaveReason.EXPLICIT));
```

Comment:
Should this be `(setting[x.value] === 'explict' || setting[x.value] === true)` Right now the `|| true` makes the entire expression always true

Reply:
hmm testing there's actually no difference in behavior since it's gate kept by `SaveReason.EXPLICIT`, but I'll definitely add in anyways to be logically correct.